add auth header (#127)
dgraph-io/dgraph-docs: content/graphql/lambda/field.md

### Schema

To set up a lambda function, first you need to define it on your GraphQL schema
by using the `@lambda` directive.

For example, to define a lambda function for the `rank` and `bio` fields in
`Author`:

```graphql
type Author {
  id: ID!
  name: String! @search(by: [hash, trigram])
  dob: DateTime @search
  reputation: Float @search
  bio: String @lambda
  rank: Int @lambda
  isMe: Boolean @lambda
}
```

You can also define `@lambda` fields on interfaces:

```graphql
interface Character {
  id: ID!
  name: String! @search(by: [exact])
  bio: String @lambda
}

type Human implements Character {
  totalCredits: Float
}

type Droid implements Character {
  primaryFunction: String
}
```

### Resolvers

Once the schema is ready, you can define your JavaScript mutation function and
add it as a resolver in your JS source code. To add the resolver you can use
either the `addGraphQLResolvers` or `addMultiParentGraphQLResolvers` methods.

{{% notice "note" %}} A Lambda Field resolver can use a combination of
`parents`, `parent`, `dql`, or `graphql` inside the function. {{% /notice %}}

> {{% notice "tip" %}} This example uses `parent` for the resolver function. You

Reviewer: Thanks for trimming the line lengths! But, note that you can't break markdown links across lines... so in the case of links, it is fine for the line to exceed 80 or 100 characters (as long as necessary to avoid link breaks).

Please fix this on lines 58 and 59

can find additional resolver examples using `dql` in the [Lambda queries
article]({{< relref "query.md" >}}), and using `graphql` in the [Lambda
mutations article]({{< relref "mutation.md" >}}). {{% /notice %}}

For example, to define JavaScript lambda functions for...

- `Author`,
- `Character`,
- `Human`, and
- `Droid`

...and add them as resolvers, do the following:

```javascript

> const authorBio = ({ parent: { name, dob } }) =>

Reviewer: It's also fine to ignore line length guidelines for syntax snippets :)

  `My name is ${name} and I was born on ${dob}.`;
const characterBio = ({ parent: { name } }) => `My name is ${name}.`;
const humanBio = ({ parent: { name, totalCredits } }) =>
  `My name is ${name}. I have ${totalCredits} credits.`;
const droidBio = ({ parent: { name, primaryFunction } }) =>
  `My name is ${name}. My primary function is ${primaryFunction}.`;

self.addGraphQLResolvers({
  "Author.bio": authorBio,
  "Character.bio": characterBio,
  "Human.bio": humanBio,
  "Droid.bio": droidBio,
});
```

> An example, adding a resolver for `rank` using a `graphql` call:

Reviewer: ```suggestion
For example, you can add a resolver for `rank` using a `graphql` call, as follows:
```

```javascript
async function rank({ parents }) {
  const idRepList = parents.map(function (parent) {
    return { id: parent.id, rep: parent.reputation };
  });
  const idRepMap = {};
  idRepList
    .sort((a, b) => (a.rep > b.rep ? -1 : 1))
    .forEach((a, i) => (idRepMap[a.id] = i + 1));
  return parents.map((p) => idRepMap[p.id]);
}

self.addMultiParentGraphQLResolvers({
  "Author.rank": rank,
});
```

> An example of using the client provided JWT to return true if the custom claim

Reviewer: ```suggestion
The following example demonstrates using the client-provided JWT to return true if the custom claim
```

> for `USER` from the JWT matches the Author's id.

Reviewer: ```suggestion
for `USER` from the JWT matches the Author's `id`:
```
